WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -75,4 +75,16 @@ def person_data
       dob: @person.dob,
     }
   end
+
+  def compute_auxiliary_data
+    @auxiliary_data_start = Timestamp.find_or_create_by(name: 'auxiliary_data_start')
+    @auxiliary_data_end = Timestamp.find_or_create_by(name: 'auxiliary_data_end')
+  end
+
+  def do_compute_auxiliary_data
+    # Reset the end date. The start date is updated directly before the computation.
+    Timestamp.find_or_create_by(name: 'auxiliary_data_end').update date: nil
+    ComputeAuxiliaryData.perform_later
```

Comment:
nit: we should always be calling `update!`, not `update`.

I just had an idea: what if we created a subclass of `ApplicationJob` called `TimedApplicationJob`? Rough idea:

```ruby
class TimedApplicationJob < ApplicationJob
  def self.start_timestamp
    Timestamp.find_or_create_by(name: "#{self.class.name.underscore}_start")
  end

  def self.end_timestamp
    Timestamp.find_or_create_by(name: "#{self.class.name.underscore}_end")
  end

  around_perform do |job, block|
    start_timestamp.touch :date
    end_timestamp.update! date: nil
    block.call
    end_timestamp.touch :date
  end
end
```

(you might also want to override `perform_later` to clear the `end_timestamp`, but that should be easy...)


edit: On further thought, this will probably have to be written using include instead of sublassing, because the `self.*` methods I wrote above are going to be defined on `TimedApplicationJob`, not on the desired subclasses such as `ComputeAuxiliaryData`.